src/memory/managed_ptr.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <cuda.h>
+
+#include <memory/allocator.hpp>
+
+namespace nest {
+namespace mc {
+namespace memory {
+
+// used to indicate that the type pointed to by the managed_ptr is to be
+// constructed in the managed_ptr constructor
+struct construct_in_place_tag {};
+
+// Like std::unique_ptr, but for CUDA managed memory.
```

Comment:
Agree, the comments are a bit thin. All this memory stuff is due a big refactor, and it has to be said that the documentation for `std::unique_ptr` is pretty good.